### PR TITLE
Closes #5

### LINKED ISSUE #5: Support regexp check

### PR DESCRIPTION
**Pull request by Charlyzzz**
Adds regular expressions support

### CODE REVIEW

File: spec/integration_spec.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
                                  extra: '')
 
     expect(response).to eq response_type: :structured,
-                           test_results: [{title: 'test1', status: :passed, result: nil}],
+                           test_results: [{ title: 'test1', status: :passed, result: nil }],
```
Comment (flbulgarelli) [inline spec/integration_spec.rb]: I'd like to see an integration test that covers regexps

File: spec/comparators/hashed_configuration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require_relative '../spec_helper'
+
+describe HashedConfiguration do
```
Comment (flbulgarelli) [inline spec/comparators/hashed_configuration_spec.rb]: I don't really believe that such low-level testing is really meaningful.

Comment (flbulgarelli) [reply spec/comparators/hashed_configuration_spec.rb]: HashedConfiguration is not a powerful, core abstraction and it is in fact just an approach to remove one line repeated code. By testing any comparator, you are actually testing this code.

File: spec/comparators/regexp_comparator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+require_relative '../spec_helper'
+
+describe RegexpComparator do
+  let(:comparator) { RegexpComparator.new('/Foo Bar/') }
```
Comment (flbulgarelli) [inline spec/comparators/regexp_comparator_spec.rb]: The `/`'s are not actually part of the regexp, but just the ruby syntax for creating them. t 

I think this test should look like this: 

```ruby
let(:comparator) { RegexpComparator.new('Foo Bar') }
```

File: lib/comparators/regexp_comparator.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    unless source =~ @expected_regexp
+      I18n.t 'expression.failure', actual: source
+    end
+  end
+
+  private
+
+  def eval_regexp(expression)
+    begin
+      eval expression
```
Comment (flbulgarelli) [inline lib/comparators/regexp_comparator.rb]: Don't use `eval`. It is slow and a security hole. Just create the regex from the string using `Regexp.new('foo.*')` or `/#{'foo.*'}`, which is also simpler.

File: lib/comparators/regexp_comparator.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    rescue
+      nil
```
Comment (flbulgarelli) [inline lib/comparators/regexp_comparator.rb]: Please raise the exception directly here.

File: lib/comparators/hashed_configuration.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module HashedConfiguration
+  def hash_configuration(opts)
```
Comment (flbulgarelli) [inline lib/comparators/hashed_configuration.rb]: If you convert this mixin into a normal abstract class, and call this method within `initialize` here, your code could be much simpler. 

For example: 

```ruby
class HashedConfiguration 
    def initialize(opts)
         parse_options(hash_configuration(opts))
   end
   ...
end      

class RegexpComparator <  HashedConfiguration
 
  def parse_options(options)
        regexp = eval_regexp(options[:expected])
        raise 'No regexp found' unless regexp.is_a? Regexp
       @expected_regexp = regexp
  end
  ...
end
```

**Review by flbulgarelli — CHANGES_REQUESTED** [review]

File: spec/comparators/hashed_configuration_spec.rb (hunk 2 above)
Comment (Charlyzzz) [reply spec/comparators/hashed_configuration_spec.rb]: Yeah, but kind of remove the necesity of branching each comparator test between a string and a hash.

Comment (Charlyzzz) [reply spec/comparators/hashed_configuration_spec.rb]: Anyway, why was that distinction made? I'm tempted to always recieve a hash

File: lib/comparators/regexp_comparator.rb (hunk 5 above)
Comment (Charlyzzz) [reply lib/comparators/regexp_comparator.rb]: Do I need to raise an especific exception or just 

~~~ruby
Regexp.new(expression)
~~~

File: spec/comparators/hashed_configuration_spec.rb (hunk 2 above)
Comment (flbulgarelli) [reply spec/comparators/hashed_configuration_spec.rb]: The distinction is done in order to make writing the test for the end-user simpler.

File: lib/comparators/regexp_comparator.rb (hunk 5 above)
Comment (flbulgarelli) [reply lib/comparators/regexp_comparator.rb]: I think raising the default exception (`RegexpError`) is fine by now

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -2,6 +2,28 @@
 require 'mumukit/bridge'
 require 'simplecov'
 
+def valid_response(test_name)
```
Comment (flbulgarelli) [inline spec/spec_helper.rb:5]: TIP: If it is going to be used only in integration test, add those methods directly to it and not to the spec helper.

**Review by flbulgarelli — APPROVED** [review]

File: spec/spec_helper.rb (hunk 7 above)
Comment (Charlyzzz) [reply spec/spec_helper.rb:5]: Good to know, thanks!

**Pull request merged**